llvm/test/CodeGen/AArch64/sve-indexed-arithmetic.ll

```diff
@@ -0,0 +1,363 @@
+  %neg.a = fneg <16 x bfloat> %ld.a
+  %res = call <16 x bfloat> @llvm.fmuladd.v16bf16(<16 x bfloat> %neg.a, <16 x bfloat> %splat.lanes, <16 x bfloat> %ld.c)
+  store <16 x bfloat> %res, ptr %c
+  ret void
+}
+
+define void @fmls_indexed_f32_256b(ptr %a, ptr %b, ptr %c) #0 {
+; CHECK-LABEL: fmls_indexed_f32_256b:
+; CHECK:       // %bb.0:
+; CHECK-NEXT:    ldr z0, [x0]
+; CHECK-NEXT:    ldr z1, [x1]
+; CHECK-NEXT:    ldr z2, [x2]
+; CHECK-NEXT:    fmls z2.s, z0.s, z1.s[3]
+; CHECK-NEXT:    str z2, [x2]
+; CHECK-NEXT:    ret
+  %ld.a = load <8 x float>, ptr %a
+  %ld.b = load <8 x float>, ptr %b
+  %ld.c = load <8 x float>, ptr %c
+  %splat.lanes = shufflevector <8 x float> %ld.b, <8 x float> poison, <8 x i32> <i32 3, i32 3, i32 3, i32 3,
+                                                                                 i32 7, i32 7, i32 7, i32 7>
+  %neg.a = fneg <8 x float> %ld.a
+  %res = call <8 x float> @llvm.fmuladd.v8f32(<8 x float> %splat.lanes, <8 x float> %neg.a, <8 x float> %ld.c)
+  store <8 x float> %res, ptr %c
+  ret void
+}
+
+define void @fmls_indexed_f64_256b_trn1(ptr %a, ptr %b, ptr %c) #0 {
+; CHECK-LABEL: fmls_indexed_f64_256b_trn1:
+; CHECK:       // %bb.0:
+; CHECK-NEXT:    ldr z0, [x0]
+; CHECK-NEXT:    ldr z1, [x1]
+; CHECK-NEXT:    ldr z2, [x2]
+; CHECK-NEXT:    fmls z2.d, z0.d, z1.d[0]
+; CHECK-NEXT:    str z2, [x2]
+; CHECK-NEXT:    ret
+  %ld.a = load <4 x double>, ptr %a
+  %ld.b = load <4 x double>, ptr %b
+  %ld.c = load <4 x double>, ptr %c
+  %splat.lanes = shufflevector <4 x double> %ld.b, <4 x double> poison, <4 x i32> <i32 0, i32 0, i32 2, i32 2>
+  %neg.a = fneg <4 x double> %ld.a
+  %res = call <4 x double> @llvm.fmuladd.v4f64(<4 x double> %splat.lanes, <4 x double> %neg.a, <4 x double> %ld.c)
+  store <4 x double> %res, ptr %c
+  ret void
+}
+
+define void @fmls_indexed_f64_256b_trn2(ptr %a, ptr %b, ptr %c) #0 {
+; CHECK-LABEL: fmls_indexed_f64_256b_trn2:
+; CHECK:       // %bb.0:
+; CHECK-NEXT:    ldr z0, [x0]
+; CHECK-NEXT:    ldr z1, [x1]
+; CHECK-NEXT:    ldr z2, [x2]
+; CHECK-NEXT:    fmls z2.d, z0.d, z1.d[1]
+; CHECK-NEXT:    str z2, [x2]
+; CHECK-NEXT:    ret
+  %ld.a = load <4 x double>, ptr %a
+  %ld.b = load <4 x double>, ptr %b
+  %ld.c = load <4 x double>, ptr %c
+  %splat.lanes = shufflevector <4 x double> %ld.b, <4 x double> poison, <4 x i32> <i32 1, i32 1, i32 3, i32 3>
+  %neg.a = fneg <4 x double> %ld.a
+  %res = call <4 x double> @llvm.fmuladd.v4f64(<4 x double> %neg.a, <4 x double> %splat.lanes, <4 x double> %ld.c)
+  store <4 x double> %res, ptr %c
+  ret void
+}
+
+declare <16 x half> @llvm.fmuladd.v16f16(<16 x half>, <16 x half>, <16 x half>);
+declare <16 x bfloat> @llvm.fmuladd.v16bf16(<16 x bfloat>, <16 x bfloat>, <16 x bfloat>);
+declare <8 x float> @llvm.fmuladd.v8f32(<8 x float>, <8 x float>, <8 x float>);
+declare <4 x double> @llvm.fmuladd.v4f64(<4 x double>, <4 x double>, <4 x double>);
+
+attributes #0 = { noinline vscale_range(2,2) "target-features"="+sve2p1,+bf16" }
```

Comment:
I think it's worth adding "+sve-b16b16" to the target features to simplify the CHECK lines.